kiwi/runtime_checker.py

```diff
@@ -546,6 +556,77 @@ def check_syslinux_installed_if_isolinux_is_used(self):
                 if not syslinux_check_file:
                     raise KiwiRuntimeError(message)
 
+    def check_dracut_module_versions_compatible_to_kiwi(self, root_dir):
+        """
+        KIWI images which makes use of kiwi dracut modules
+        has to use module versions compatible with the version
+        of this KIWI builder code base. This is important to avoid
+        inconsistencies between the way how kiwi includes its own
+        dracut modules and former version of those dracut modules
+        which could be no longer compatible with the builder.
+        Therefore this runtime check maintains a min_version constraint
+        for which we know this KIWI builder to be compatible with.
+        """
+        message = dedent('''\n
+            Incompatible dracut-kiwi module(s) found
+
+            The image was build with KIWI version={0}. The system
+            root tree has the following dracut-kiwi-* module packages
+            installed which are too old to work with this version of KIWI.
+            Please make sure to use dracut-kiwi-* module packages
+            which are >= than the versions listed below.
+
+            {1}
+        ''')
+        kiwi_dracut_modules = {
+            '90kiwi-dump': dracut_module_type(
+                'dracut-kiwi-oem-dump', '9.20.1'
+            ),
+            '90kiwi-live': dracut_module_type(
+                'dracut-kiwi-live', '9.20.1'
+            ),
+            '90kiwi-overlay': dracut_module_type(
+                'dracut-kiwi-overlay', '9.20.1'
+            ),
+            '90kiwi-repart': dracut_module_type(
+                'dracut-kiwi-oem-repart', '9.20.1'
+            ),
+            '99kiwi-dump-reboot': dracut_module_type(
+                'dracut-kiwi-oem-dump', '9.20.1'
+            ),
+            '99kiwi-lib': dracut_module_type(
+                'dracut-kiwi-lib', '9.20.1'
+            )
+        }
+        dracut_module_dir = os.sep.join(
+            [root_dir, '/usr/lib/dracut/modules.d']
+        )
+        incompatible_modules = {}
+        for module in os.listdir(dracut_module_dir):
+            module_meta = kiwi_dracut_modules.get(module)
+            if module_meta:
+                module_version = self._get_dracut_module_version_from_pdb(
+                    self.xml_state.get_package_manager(),
+                    module_meta.package, root_dir
+                )
+                if module_version:
+                    module_version_nr = int(
+                        module_version.replace('.', '')
```

Comment:
> mmm isn't it too simple? What about 9.2.2 vs 9.1.10, 

Oh man you are so right. Thanks much for catching this. I added a commit to fix this. I know tuples can be compared and wondered about your idea using a set() and if that works too. The comparison of the set however did not work for me so I used tuples from which I know that it can be used for the operator compare